master/buildbot/process/buildstep.py

```diff
@@ -1008,8 +1010,8 @@ def checkDisconnect(self, f):
 
     def commandComplete(self, cmd):
         # Now that the build is complete, add any URLs that the buildstep had to the buildstep's status
-        for url_key in self.urls:
-            self.addURL(url_key, self.urls[url_key])
+        for url in self.urls:
+            self.addURL(url["name"], url["url"])
```

Comment:
I considered it, but decided that being able to put the name and URL as simple variables is much more useful for the function than having to convert the name/url to a dictionary of a specific format.

I could also make a second addURL function, one that takes a single parameter, but considering this is the only place that that'd be used, I think that's too excessive.